server/util_test.go

```diff
@@ -400,26 +402,69 @@ func TestInterceptConfigsWithBadPermissions(t *testing.T) {
 		t.Fatalf("Failed to catch permissions error, got: [%T] %v", err, err)
 	}
 }
-
+ 
 func TestStartCmdwithEmptyandNonEmptyMinGasPrices(t *testing.T){
-
-	// first case - raise an error
+	
 	tempDir := t.TempDir()
-	cmd := StartCmd(nil, tempDir)
-
-
-	// MinGasPrices ="0.22  -> no error raised
-	cfg := config.DefaultConfig()
-	cfg.BaseConfig.MinGasPrices ="0.22"
+	
+	var app types.AppCreator
 
-	cmd.PreRunE = preRunETestImpl
+	// TODO fix panic on app := appCreator(ctx.Logger, db, traceWriter, ctx.Viper) in func startInProcess(ctx *Context, clientCtx client.Context, appCreator types.AppCreator) 
+	cmd := StartCmd(app, tempDir)  
```

Comment:
Unable to set arguments to `app types.AppCreator` to remove panic [here](https://github.com/cosmos/cosmos-sdk/blob/master/server/start.go#L248) . The  full test output can be found [here](https://play.golang.org/p/lWozD048jLD). Just wonder what is difference between `startInProcess(cfg Config, val *Validator) error` and `startInProcess(ctx *Context, clientCtx client.Context, appCreator types.AppCreator) error`. I guess the first one is used for testing. Right? cc @amaurym

Reply:
Arguments in `app' are set within `startInProcess` so I can not set them externally for testing purposes.